projects/media-viewer/src/lib/toolbar/left-pane/left-pane.component.ts

```diff
@@ -18,10 +19,41 @@ export class ToolbarLeftPaneComponent {
   @Input() showNavigationBtns: boolean;
   @Input() sidebarOpen: BehaviorSubject<boolean>;
   @Input() searchBarHidden: BehaviorSubject<boolean>;
-  @Input() drawMode: BehaviorSubject<boolean>;
-  @Input() highlightMode: BehaviorSubject<boolean>;
+  // Local presentation state that can be pre-initialised from the template
+  @Input() highlightMode = false;
+  @Input() drawMode = false;
+
+  // local array of any subscriptions so that we can tidy them up later
+  private subscriptions: Subscription[] = [];
+  constructor(private readonly toolbarEventsService: ToolbarEventsService) {}
+
+  ngOnInit(): void {
+    // Listen for any changes invoked on the toolbar events Service and initialise any default behaviour state
+    this.subscriptions.push(this.toolbarEventsService.highlightMode.subscribe((toggleValue) => {
+      this.highlightMode = toggleValue;
+    }));
+    this.subscriptions.push(this.toolbarEventsService.drawMode.subscribe((toggleValue) => {
+      this.drawMode = toggleValue;
+    }));
+  }
 
-  constructor() {}
+  ngOnDestroy(): void {
+    // Clean up any subscriptions that we may have
+    this.subscriptions.forEach((subscription) => {
```

Comment:
I mentioned this to Taleb the other day - I prefer to use the `for (const subscription of this.subscriptions) { ... }` approach. There is not much between them but I prefer to use a language construct rather than a higher order function. It feels more natural.

Reply:
@linusnorton changed to your suggestion. Can you re-check.
Overall, i’m not too fussed about developer styles within the team on choice of control flow. Ultimately there are numerous ways of iterating and it comes down to preference. Maybe we need to look at a code linter config so that we pick these up as we right if its deemed important.

Reply:
Was just adding my preference but it doesn't matter ultimately.